SCClassLibrary/Common/Core/Thread.sc

```diff
@@ -110,6 +111,27 @@ Routine : Thread {
 		_RoutineResume
 		^this.primitiveFailed
 	}
+	reschedule { arg argClock, quant;
+		var doResched = {
+			// Thread:isPlaying only answers if the thread is waiting
+			// It *doesn't* confirm that it is actually scheduled on a clock
+			if(this.nextBeat.isNil) {
+				Error("% can't be rescheduled when idle; use 'play' instead".format(this.class.name)).throw;
+			};
+			rescheduledTime = quant.asQuant.nextTimeOnGrid(clock, this.nextBeat);
+			if(argClock.isNil) { argClock = clock };
+			if(argClock !== clock) {
+				// convert to new clock's time
+				rescheduledTime = argClock.secs2beats(clock.beats2secs(rescheduledTime));
+			};
+			clock = argClock;
+		};
+		if(this === thisThread) {
```

Comment:
if there is a good name for what happens here, I think this should be a method on function, similar to `forkIfNeeded`

Reply:
How would it look? If it's a method of Function, then you wouldn't have implicit access to the thread as `this`, so maybe it would have to be like this, in context of this PR: `{ ... }.deferFromThread(this/*.threadPlayer*/)`. (I'm not sure this is the place to bikeshed a new method though.)

Reply:
>I'm not sure this is the place to bikeshed a new method though

As a non native speaker, the meaning of bikeshedding remains obscure (I'm pretty sure I've never done it, although I do have a bicycle ;-)). 


Joke aside, this definitely is the place to discuss how to factor a problem. This may include anew method, and the general usefulness of such a method. If it turns out to be a large discussion, we can still do this in a separate thread.

I think your second version is good, if we assume that the intention is to _defer away from_ or _defer off_  a given thread. This method makes clear what reading the code leaves obscure.

Reply:
OK, I've done that (and even instinctively named the method `deferAwayFrom`).

I have to admit, writing the documentation for the new method renewed my doubts about it (though its usages in both `reschedule` implementations do look cleaner). But, let's leave that to the review process.